src/fragmentarium/ui/fragment/TeiExport.js

```diff
@@ -0,0 +1,138 @@
+export function teiExport(fragment) {
+  var res = ''
+  const header = getHeader(fragment)
+  const body = getBody(fragment)
+  const end = getEnd()
+
+  res += header
+  res += body
+  res += end
+
+  console.log(fragment)
+
+  return prettifyXml(res)
+}
+
+function getHeader(fragment) {
+  var res = ''
+  const start =
+    '<?xml version="1.0" encoding="UTF-8"?><TEI xmlns="http://www.tei-c.org/ns/1.0"><teiHeader><fileDesc>'
+  const title_stmt =
+    '<titleStmt><title>' + fragment.number + '</title></titleStmt>'
+  const pub_stmt =
+    '<publicationStmt><p>' +
+    fragment.museum.name +
+    '</p><p>' +
+    fragment.collection +
+    ' Collection</p></publicationStmt>'
+
+  var source_desc = '<sourceDesc><p>' + fragment.description + '</p>'
+
+  if (fragment.publication)
+    source_desc += '<p>Publication: ' + fragment.publication + '</p>'
+  // if (fragment.references.length>0)  source_desc += getBibiliography(fragment);
+
+  const end = '</sourceDesc></fileDesc></teiHeader><text><body>'
+
+  return res + start + title_stmt + pub_stmt + source_desc + end
+}
+
+// function getBibiliography(fragment){
+// var res = '<biblFull>'
+
+// Object.values(fragment.references).forEach(val => {
+//  res+=addSingleReference(val);
+// })
+
+// res+= '</biblFull>'
+// return res
+// }
+
+// function addSingleReference(ref){
+// var res = '<titleStmt>';
+// res += '<title>'+ ref.document.cslData.title + '</title>';
+
+// var short_author = ref.document.cslData['container-title-short']
+// var author = ref.document.cslData.author
+
+// if(author || short_author) {
+//   res += '<author>'
+//     if(short_author)res += short_author
+//     else res += author[0].given + " "+author[0].family
+//   res += '</author>'
+// }
+
+// res += '</titleStmt>'
+// return res
+// }
+
+function getEnd() {
+  return '</body></text></TEI>'
+}
+
+function getBody(fragment) {
+  return fragment.text.allLines.filter(isNotEmpty()).map(getParagraph).join('')
+}
+
+function isNotEmpty() {
+  return (line) => line.type !== 'EmptyLine'
```

Comment:
There should be no need to return another function. See also type guard below.

Reply:
added notIsEmptyLine(line: AbstractLine) to type-guards and removed function

Reply:
now using isEmtpyLine